src/datasources/human-description-api/human-description-api.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IHumanDescriptionApi } from '../../domain/interfaces/human-description-api.interface';
+import {
+  MessageTemplates,
+  HumanReadableFragment,
+  ValueType,
+  Expression,
+} from './entities/human-description.entity';
+
+const isValueType = (type: unknown): type is ValueType => {
+  return Object.values(ValueType).includes(type as ValueType);
+};
+
+@Injectable()
+export class HumanDescriptionApi implements IHumanDescriptionApi {
+  private readonly parsedMessages: MessageTemplates;
+
+  /**
+   *  Regex Template that matches two patterns
+   *  1. Double curly braces consisting of 2 groups separated by a space
+   *  2. Any non-whitespace character i.e. simple words
+   */
+  private readonly templateRegex = /{{(.*?)\s(\$.*?)}}|(\S+)/g;
+
+  constructor(
+    @Inject('ContractDescriptions')
+    private readonly contractDescriptions: Expression,
+  ) {
+    this.parsedMessages = this.parseMessages(this.contractDescriptions);
```

Comment:
Not sure I understand the question, do you mean creating an additional class to handle the parsing? It could be nice, but no strong opinion on that from my side.

Reply:
I agree @iamacook . I think the parsed messages can be provided via DI in a ready-to-consume manner to `HumanDescriptionApi`. You can use a Nest Factory to provide those.

Reply:
I've simplified `HumanDescriptionApi` a lot now. It only has one get function and essentially returns the json. Once we move it to an external source we can define more properties on the class like `baseUrl` etc.